cmd/ingester/app/processor/span_processor.go

```diff
@@ -58,9 +59,9 @@ func NewSpanProcessor(params SpanProcessorParams) *KafkaSpanProcessor {
 
 // Process unmarshals and writes a single kafka message
 func (s KafkaSpanProcessor) Process(message Message) error {
-	mSpan, err := s.unmarshaller.Unmarshal(message.Value())
+	span, err := s.unmarshaller.Unmarshal(message.Value())
 	if err != nil {
 		return fmt.Errorf("cannot unmarshall byte array into span: %w", err)
 	}
-	return s.writer.WriteSpan(mSpan)
+	return s.writer.WriteSpan(context.TODO(), span)
```

Comment:
Using TODO here because KafkaSpanProcessor should be managing the context earlier.